lib/collection/certificate.js

```diff
@@ -0,0 +1,129 @@
+var _ = require('../util').lodash,
+    Property = require('./property').Property,
+    UrlMatchPatternList = require('../url-pattern/url-match-pattern-list').UrlMatchPatternList,
+
+    Certificate;
+
+/**
+* The following is the object representation accepted as param for the Certificate constructor.
+* Also the type of the object returned when {@link Property#toJSON} or {@link Property#toObjectResolved} is called on a
+* Certificate instance.
+* @typedef Certificate~definition
+* @property {String} [name] A name for the certificate
+* @property {Array} [matches] A list of match patterns
+* @property {String} [keyPath] The path on the file system to the file that contains the key
+* @property {String} [pemPath] The path on the file system to the certificate file
+* @property {String} [passphrase] The passphrase for the certificate key
+*
+* @example <caption>JSON definition of an example certificate object</caption>
+* {
+*     name: 'My certificate for example.com',
+*     matches: ['https://example.com/*'],
+*     key: { src: '/User/path/to/certificate/key' },
+*     cert: { src: '/User/path/to/certificate' },
+*     passphrase: 'iampassphrase'
```

Comment:
@kamalaknn @SamvelRaja - any reply to this? I don't have much context, but want to ensure all conversations are closed.